hash body only when POST and body not empty

ynohat opened the pull request:
This is not required by the spec and yields a "signature does not match" (401) response.

ynohat committed:
don't hash body when signing

This is not required by the spec and yields a "401 signature does not match" response.

chonthu commented:
https://developer.akamai.com/introduction/Client_Auth.html For post request its seems this is necessary. The 401 goes away when this was removed for your case? I use this in production as is just fine, can you show me example of the request your making to the api?

ynohat commented:
You are quite right, upon re-reading the spec the hash is necessary. However removing the hash does make the following case work, with the hash I get a 401.

```
package main

import (
	"log"
	"os"

	"io/ioutil"

	"github.com/chonthu/go-edgegrid"
)

func main() {
	api := edgegrid.New(
		os.Getenv("FASTDNS_API_HOST"),
		os.Getenv("FASTDNS_API_CLIENTTOKEN"),
		os.Getenv("FASTDNS_API_CLIENTSECRET"),
		os.Getenv("FASTDNS_API_ACCESSTOKEN"),
		"",
	)

	resp, err := api.Send("GET", "/config-dns/v1/zones/example.com", "")

	if err != nil {
		log.Panicln(err)
	}

	defer resp.Body.Close()

	body, _ := ioutil.ReadAll(resp.Body)
	bodyStr := string(body[:])
	log.Println(bodyStr)
}
```

Maybe the problem is that if no body is present in the request, the data to sign is expected to be the empty string?

Do you use this code in production in contexts without bodies?

ynohat committed:
fix body hash logic

The spec says that the body should only be hashed for POST requests and that if the body is empty, the hash should be empty as well (but the tab delimiter should be preserved).

ynohat commented:
@chonthu I believe both use cases should now be addressed by this PR

chonthu commented:
Looks good to me, thanks for the PR
